v2/internal/metrics/metrics.go

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package metrics
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+const (
+	putLabel    = "PUT"
+	deleteLabel = "DELETE"
+)
+
+// TODO: Add more status codes here if there are more expected codes.
+// Basic http API request status codes
+var statusCodes = []float64{
```

Comment:
Are these actually floats or are they ints?

Reply:
They are ints, they get parsed as floats. However a histogram doesn't make sense here, I'm actually planning to remove this and merge status code as a label with `request_counter`